docker/postgres/Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM postgres:11
+
+# Add db init script to install OSM-specific Postgres functions/extensions.
+ADD docker/postgres/openstreetmap-postgres-init.sh /docker-entrypoint-initdb.d/
+
+# Custom database functions are in a SQL file.
+ADD db/functions/functions.sql /usr/local/sbin/osm-db-functions.sql
```

Comment:
A couple of comments: 

* /usr/local/sbin seems to be a somewhat unusual location for sql scripts. 

* Also, I'm not 100% sure if we need this script at all. At least maptile_for_point (for the /changes endpoint) and xid_to_int4 (for the old osmosis based replication) are likely candidates to be removed in the near future. tile_for_point functionality is probably part of the quad_tile, and might also no longer be needed.

Reply:
Do you have an alternate suggestion for a location?

Tests fail when these functions are excluded.

Reply:
Do you recall which test did in fact fail?

Regarding the location, can't you use /app/db/functions/functions.sql as is?

Reply:
Here’s the failing test suite that was fixed by adding functions.sql: https://github.com/migurski/openstreetmap-website/runs/1619876640?check_suite_focus=true

We don’t use `/app` in the DB container, so it’s not there by default.

Reply:
Ah, yes, right, the script need to go to another container. Assumption confirmed, it was the /changes endpoint. I really want to get rid of this code, as it's outdated and not used by anyone except for some single unknown user. (-> re-opened #2486). Once this is done, we don't need the sql script anymore.

```
 Failure:
Api::ChangesControllerTest#test_changes_hours_valid [/app/test/controllers/api/changes_controller_test.rb:91]:
Expected response to be a <2XX: success>, but was a <500: Internal Server Error>
Response body: ActiveRecord::StatementInvalid: PG::UndefinedFunction: ERROR:  function maptile_for_point(integer, integer, integer) does not exist
LINE 1: SELECT COUNT(*) AS count_all, maptile_for_point(latitude, lo...
```

Reply:
I'm sorry but I agree that `/usr/local/sbin` is daft - it's not an executable and even if it was why sbin and not bin? I'd suggest lib or share for this.

Reply:
Now that the /changes endpoint has been removed, do we need the SQL script at all?

Reply:
Unfortunately yes, we still need it. The `tile_for_point` SQL function is not needed for day-to-day operations (and so not required for the tests), since we calculate the tile using the QuadTile gem before storing it, and we again use the QuadTile gem to generate the (giant) select queries.

But there's also the database migrations where the columns were added, and those migrations backfill the tile columns using the SQL function. 

I have much else to say about ancient migrations, but for now, we can't just drop the SQL function entirely.

Reply:
That's a bit odd that you need those migrations on an empty database inside a container. Is there some way to start fresh with a more modern state of the database sql scripts, and skip any ancient stuff?

Reply:
To answer your second question, yes, you run those old migrations - the instructions have `docker-compose run --rm web bundle exec rake db:migrate`

But the wider topic is whether we should keep those old migrations, whether we see the database defined by either a) the sum of all the migrations or b) the structure.sql file (opinions vary), and finally whether if we do drop old migrations should we have a 'fake' starter migration, or should we just keep a certain number of migrations and change all the documentation to use rake db:structure:load (or similar), and finally (finally) if we don't keep all the migrations, how do we ensure that the structure file matches the sum of the migrations that have been run production. So that's not a simple set of decisions.

Reply:
Somehow I assumed a new database would be set up using db/structure.sql, and then the contents of the migrations table would define, which additional migrations would be needed, by comparing that table's contents against what's inside db/migrate. Maybe the mechanism works in a completely different way, but then I don't really know what's the purpose of that db/structure.sql is in the first place (sorry, I know this all sounds much like noob questions).

Reply:
In any case, running the migration passes on a fresh install even without the db scripts, so I'm even more confused now why they would be needed.

Reply:
Let's discuss this on a new issue - otherwise this conversation will never be found!